apps/mobile/src/queries/query.ts

```diff
@@ -1,9 +1,25 @@
 import { QueryClient } from '@tanstack/react-query';
+import { HttpStatusCode, isAxiosError } from 'axios';
+
+const RETRY_LIMIT = 5;
```

Comment:
If we stop re-trying, we should show a callout to the users to let them know, like this one for `Balances` 

![Screenshot 2025-05-14 at 05 49 52](https://github.com/user-attachments/assets/73062a42-0d30-405f-9362-7fc656c75081)

Otherwise they won't know we aren't fetching anymore. We also need to add more error states to let users initiate re-fetch more easily. 

I think we should bundle this change with [LEA-2562](https://linear.app/leather-io/issue/LEA-2562/error-boundary)